assets/js/theme/global/cart-preview.js

```diff
@@ -49,4 +51,53 @@ export default function () {
                 .hide();
         });
     });
+
+    // Check if we can use localStorage
+    function storageAvailable(type) {
+        const storage = window[type];
+        try {
+            const x = '__storage_test__';
+            storage.setItem(x, x);
+            storage.removeItem(x);
+            return true;
+        } catch (e) {
+            return e instanceof DOMException && (
+                // everything except Firefox
+                e.code === 22 ||
+                // Firefox
+                e.code === 1014 ||
+                // test name field too, because code might not be present
+                // everything except Firefox
+                e.name === 'QuotaExceededError' ||
+                // Firefox
+                e.name === 'NS_ERROR_DOM_QUOTA_REACHED') &&
+                // acknowledge QuotaExceededError only if there's something already stored
+                storage.length !== 0;
+        }
+    }
+
+    let quantity = 0;
+
+    // Get existing quantity from localStorage if found
+    if (storageAvailable('localStorage')) {
+        if (localStorage.getItem('cart-quantity')) {
+            quantity = Number(localStorage.getItem('cart-quantity'));
+            $body.trigger('cart-quantity-update', quantity);
+        }
+    }
+
+    // Get updated cart quantity from the Cart API
+    const cartQtyPromise = new Promise(resolve => {
```

Comment:
I just noticed that the `getCartQuantity` function in stencil-utils does not handle api errors properly.

https://github.com/bigcommerce/stencil-utils/blob/master/src/api/cart.js#L26 should be modified to check for an error and then call `callback(err)`. When successful, it should call `callback(null, quantity)`. The standard for JS callback functions is that the first param the error and the second param is the result.

Once that is done, this PR should be modified to handle the error and call the reject function in the Promise.

Reply:
https://github.com/bigcommerce/stencil-utils/pull/88